docs/source/notes/libtorch_stable_abi.md

```diff
@@ -0,0 +1,48 @@
+# LibTorch Stable ABI
+
+This note will eventually contain more details on how to use the APIs in torch/csrc/stable. For the moment, it contains a table of internal representations:
+1. type in custom extension: type used within the end user custom library.
+2. StableIValue representation: a stable conversion of the type to liaison between the user model vs libtorch.so.
+3. type in libtorch: type used within libtorch.so.
+4. Schema Type: type as described by the schema, which we hail as the source of truth for both ATen ops in native_functions.yaml and for user defined custom ops registered to the dispatcher.
+
+|  type in custom extension    |   StableIValue representation   |   type in libtorch  |   Schema Type  |
+| -------- | ------- | ------- | ------- |
+| std::optional\<S> | \*reinterpret_cast\<(StableIValue\*)\*>, pointer to a StableIValue recursively defined | std::optional\<T> | OptionalType |
+| std::nullopt | \*reinterpret_cast\<nullptr_t\*> | IValue() | NoneType |
+| RAIIATH | \*reinterpret_cast\<uint64_t\*> of AtenTensorHandle | at::Tensor |  TensorType |
+| int32_t | \*reinterpret_cast\<uint64_t\*> | at::ScalarType | ScalarTypeType |
+| int32_t | \*reinterpret_cast\<uint64_t\*> | at::Layout | LayoutType |
+| int32_t | \*reinterpret_cast\<uint64_t\*> | at::MemoryFormat | MemoryFormatType |
+| bool | \*reinterpret_cast\<uint64_t\*> | bool | BoolType |
+| int64_t | \*reinterpret_cast\<uint64_t\*> | int64_t | IntType |
+| float | \*reinterpret_cast\<uint64_t\*> | double | FloatType |
+| ? | ? | c10::Device | DeviceObjType |
```

Comment:
You did define the StableIValue for this one by side effect in this PR.
We actually define this for any object <64bits, should we say it?